validphys2/src/validphys/closuretest/inconsistent_closuretest/multiclosure_inconsistent.py

```diff
@@ -0,0 +1,209 @@
+"""
+closuretest/inconsistent_closuretest/multiclosure_inconsistent.py
+
+Module containing all of the statistical estimators which are
+averaged across multiple inconsistent fits. The actions
+in this module are used to produce results which are plotted in
+``multiclosure_inconsistent_output.py``
+
+"""
+
+import numpy as np
+import dataclasses
+
+from validphys import covmats
+from validphys.calcutils import calc_chi2
+from validphys.results import ThPredictionsResult
+from validphys.closuretest.closure_checks import check_multifit_replicas
+
+from reportengine import collect
+
+
+@dataclasses.dataclass(frozen=True)
+class PCAInternalMulticlosureLoader:
+    """
+    Parameters
+    ----------
+    closures_th: list
+        list containing validphys.results.ThPredictionsResult objects
+        for each fit
+
+    law_th: ThPredictionsResult object
+        underlying law theory predictions
+
+    pc_basis: np.array
+        basis of principal components
+
+    n_comp: int
+        number of principal components kept after regularisation
+
+    covmat_pca: np.array
+        regularised covariance matrix computed from replicas
+        of theory predictions
+
+    sqrt_covmat_pca: np.array
+        cholesky decomposed covariance matrix
+    """
+
+    closures_th: list
+    law_th: ThPredictionsResult
+    pc_basis: np.array
+    n_comp: int
+    covmat_pca: np.array
+    sqrt_covmat_pca: np.array
+
+
+@check_multifit_replicas
+def internal_multiclosure_dataset_loader_pca(
```

Comment:
In general this function is too big for my taste, maybe it could be splitted in some more unit functions